primeqa/pipelines/components/retriever/sparse.py

```diff
@@ -29,22 +31,55 @@ class BM25Retriever(RetrieverComponent):
             "description": "Path to root directory where index is stored",
         },
     )
-    index_name: str = field(
+
+    max_num_documents: int = field(
+        default=5,
         metadata={
-            "name": "Index name",
+            "name": "Maximum number of retrieved documents",
+            "range": [1, 100, 1],
+            "api_support": True,
+            "exclude_from_hash": True,
         },
     )
-    max_num_documents: int = field(
-        default=5,
-        metadata={"name": "Maximum number of documents", "range": [1, 100, 1]},
+    
+    num_workers: int = field(
+        default=1,
+        metadata={
+            "name": "Num worker threads",
+            "range": [1, 100, 1],
+            "exclude_from_hash": True,
+        },
     )
 
     def __post_init__(self):
         # Placeholder variables
+        self._index_path=f"{self.index_root}/{self.index_name}"
         self._searcher = None
+        
+    def __hash__(self) -> int:
+        # Step 1: Identify all fields to be included in the hash
+        hashable_fields = [
+            k
+            for k, v in self.__class__.__dataclass_fields__.items()
+            if not "exclude_from_hash" in v.metadata
+            or not v.metadata["exclude_from_hash"]
+        ]
+
+        # Step 2: Run
+        return hash(
+            f"{self.__class__.__name__}::{json.dumps({k: v for k, v in vars(self).items() if k in hashable_fields}, sort_keys=True)}"
+        )
 
     def load(self, *args, **kwargs):
-        pass
+        self._searcher = PyseriniRetriever(self._index_path)
 
     def retrieve(self, input_texts: List[str], *args, **kwargs):
-        pass
+        qids = [str(idx) for  idx, query in enumerate(input_texts) ]
```

Comment:
Shouldn't this change to `queries` as per change in the `base.py` for `retriever` component?